tests/integration-tests/tests/common/utils.py

```diff
@@ -231,3 +235,33 @@ def get_route_tables(subnet_id, region):
         Filters=[{"Name": "association.subnet-id", "Values": [subnet_id]}]
     )
     return [table["RouteTableId"] for table in response["RouteTables"]]
+
+
+def gather_system_report(cluster, request, scheduler_commands, remote_command_executor):
+    out_dir = request.config.getoption("output_dir")
+    logging.info(f"Get system information about test: {request.node.name}")
+    logging.info(f"Run command on head node to get system information about test: {request.node.name}")
+    result = remote_command_executor.run_remote_script(SYSTEM_REPORT_SCRIPT, args=["/tmp"])
+    logging.debug(f"result.failed={result.failed}")
+    logging.debug(f"result.stdout={result.stdout}")
+    logging.info(f"Copy results from remote cluster into: {out_dir}/system_report/systemreport_compute_node_{request.node.name}.tar.gz")
+    os.makedirs(f"{out_dir}/system_report", exist_ok=True)
+    remote_command_executor.get_remote_files(
+        "/tmp/systemreport.tar.gz",
+        f"{out_dir}/system_report/systemreport_head_node_{request.node.name}.tar.gz",
+        preserve_mode=False,
+    )
+    logging.info("Copy head node system report: successfully executed")
+
+    logging.info(f"Run command on a compute node to get system information about test: {request.node.name}")
+    result = scheduler_commands.submit_script(SYSTEM_REPORT_SCRIPT, script_args=["/opt/parallelcluster/shared"])
+    job_id = scheduler_commands.assert_job_submitted(result.stdout)
+    scheduler_commands.wait_job_completed(job_id)
+    scheduler_commands.assert_job_succeeded(job_id)
+    logging.info(f"Copy results from remote cluster into: {out_dir}/system_report/systemreport_compute_node_{request.node.name}.tar.gz")
+    remote_command_executor.get_remote_files(
+        "/opt/parallelcluster/shared/systemreport.tar.gz",
```

Comment:
same here: `/opt/parallelcluster/shared` is used in more than 1 place, let's define a variable for it, to avoid typos.

Reply:
Done